subsys/logging/Kconfig.formatting

```diff
@@ -174,9 +174,24 @@ config LOG_BACKEND_FORMAT_TIMESTAMP
 	bool "Timestamp formatting in the backend"
 	depends on LOG_BACKEND_SUPPORTS_FORMAT_TIMESTAMP
 	default y
+	help
+	  When enabled timestamp is formatted.
+
+choice LOG_BACKEND_FORMAT_TIMESTAMP_MODE
+	prompt "Timestamp format mode"
+	default LOG_OUTPUT_FORMAT_DATE_TIMESTAMP if LOG_TIMESTAMP_USE_REALTIME
+	default LOG_OUTPUT_FORMAT_TIME_TIMESTAMP
+
+config LOG_OUTPUT_FORMAT_TIME_TIMESTAMP
+	bool "Format timestamp in time format"
 	help
 	  When enabled timestamp is formatted to hh:mm:ss:ms,us.
 
+config LOG_OUTPUT_FORMAT_DATE_TIMESTAMP
+	bool "Format timestamp in date format"
+	help
+	  When enabled timestamp is formatted to YYYY-MM-DD hh:mm:ss:ms,us.
```

Comment:
Would it make sense to use `T` as a delimiter instead of space between date and time stamp? And how about showing fractions of seconds using 6 decimal numbers?

Like this: `YYYY-MM-DDThh:mm:ss.SSSUUU`, or `2024-09-12T15:47:36.123456`. The dot could be comma as well.

Reasoning: then the timestamp would look more like common ISO 8601 timestamp.